### DISCUSSION TITLE
Add support for remote spec. or rulesets

### DISCUSSION THREAD

**Discussion by Herve-M** [daveshanley/vacuum#336] [Ideas]
Hello :-) 

Would be nice to be able to use `vacuum` with remote spec. or rulesets like:
` vacuum lint -r https://raw.githubusercontent.com/azure/azure-api-style-guide/main/spectral.yaml https://raw.githubusercontent.com/pb33f/openapi-changes/main/sample-specs/petstorev3.json`

Thanks,

**Comment by daveshanley** [comment]
This feature is coming soon. Watch this space.

**Comment by yashwason** [comment]
Hi,
I have a similar requirement and believe it might be related to this discussion. If not, I can open a separate thread.

I'm currently in the process of splitting a large OpenAPI spec into smaller files hosted on a remote URL and referenced from the index file.

I'm interested in including the content of these smaller files in linting. For instance, if a remote schema lacks a description while a linting rule mandates it, I'd like to receive a warning.

It seems like there might be some overlap between my idea and the one originally suggested here. I guess the core utilities required for my idea and the one originally suggested here are somewhat related?

I'd appreciate any feedback on whether this could be considered for the roadmap.

Vacuum rocks @daveshanley!

**Reply by daveshanley** [reply]
Thank you for the kind words! 

I am not quite sure what you're referring to here, but vacuum and libopenapi both support exploded specs with local or remote references across many files. 

Rulesets are now expandable and sharable too, so hopefully that covers all your needs, let me know if not.

**Reply by yashwason** [reply]
In other words, I would like to _lint the remote specs too;_ and get a warning for rule violations **in them**.

For example, if I am referring to a schema from a remote spec and it lacks a description; I'd like to receive a warning if I have the `component-description` rule enabled for vacuum.

**Comment by daveshanley** [comment]
I forgot to update this thread, this feature is available!

https://quobix.com/vacuum/rulesets/sharing/
